src/incubator/TextField/Label.tsx

```diff
@@ -0,0 +1,43 @@
+import React, {useContext} from 'react';
+import {TextStyle} from 'react-native';
+import {Colors} from '../../style';
+import Text, {TextPropTypes} from '../../components/text';
+import {ColorType, ValidationMessagePosition} from './types';
+import {getColorByState} from './Presenter';
+import FieldContext from './FieldContext';
+
+export interface LabelProps {
+  label?: string;
```

Comment:
Few things: 1) is 'label' is the best name for the COMPONENT? It is the 'title', right? The name 'title' gives the user a better clue to what this component is (a display of some text on top), while 'label' is very generic and a bit confusing in my opinion, like where this label is being displayed?
2) The 'label' PROP name - also a bit confusing as 'label' is the component, and what you're asking for is a string. I think 'text' is more suitable here.
3) Shouldn't the 'label' prop be required (not optional)?

Reply:
Yes. `label` is the right name for this prop. I'm personally more familiar with `label` for this element. 
I verified with Yulia and the original guidelines used `label`. 
And the guidelines I used to structure the component also use this name
https://uxdesign.cc/text-fields-forms-design-ui-components-series-2b32b2beebd0

Last, most important, this is the name people use for it in the community, see here: 
https://callstack.github.io/react-native-paper/text-input.html
https://react-native-elements.github.io/react-native-elements/docs/input#props

Regarding the component name, I just name it like the prop (like in all the other internal components). It's an internal component, why is that a problem?

Regarding
> and what you're asking for is a string. I think 'text' is more suitable here.

Yes, I understand what you mean. The reason for that is so I can extend the TextField types nicely and use all the internal component to get the typings. 
This way  (almost) each component responsible to declare its part of the API

3) No. not all fields require label. You don't have to render it if you don't want to.